src/xrpld/app/tx/detail/PayChan.cpp

```diff
@@ -453,16 +454,33 @@ PayChanClaim::preflight(PreflightContext const& ctx)
             return temBAD_SIGNATURE;
     }
 
+    if (auto const err = credentials::check(ctx); !isTesSuccess(err))
+        return err;
+
     return preflight2(ctx);
 }
 
+TER
+PayChanClaim::preclaim(PreclaimContext const& ctx)
+{
+    Keylet const k(ltPAYCHAN, ctx.tx[sfChannel]);
+    auto const slep = ctx.view.read(k);
+    if (!slep)
+        return tecNO_TARGET;
+
+    if (auto const err = credentials::valid(
+            ctx, ctx.tx[sfAccount], slep->getAccountID(sfDestination));
+        !isTesSuccess(err))
+        return err;
+
+    return tesSUCCESS;
+}
+
 TER
 PayChanClaim::doApply()
 {
     Keylet const k(ltPAYCHAN, ctx_.tx[sfChannel]);
     auto const slep = ctx_.view().peek(k);
-    if (!slep)
```

Comment:
Why was this removed?

Reply:
It checked in preclaim.  I'll restore it, we discuss this with Greg